.github/workflows/pulp_images.yml

```diff
@@ -11,14 +14,115 @@ env:
+        with:
+          # by default, it uses a depth of 1
+          # this fetches all history so that we can read each commit
+          fetch-depth: 0
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v1
+        with:
+          python-version: '3.8'
+      - name: Check commit message
+        if: github.event_name == 'pull_request'
+        env:
+          GITHUB_CONTEXT: ${{ github.event.pull_request.commits_url }}
+        run: |
+          echo ::group::REQUESTS
+          pip install pygithub
+          echo ::endgroup::
+          for sha in $(curl $GITHUB_CONTEXT | jq '.[].sha' | sed 's/"//g')
+          do
+            python .ci/scripts/validate_commit_message.py $sha
+            VALUE=$?
+            if [ "$VALUE" -gt 0 ]; then
+              exit $VALUE
+            fi
+          done
+        shell: bash
+  base-images:
+    needs: lint
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+    outputs:
+      image_variants: "${{ steps.image_variants.outputs.image_variants }}"
+    steps:
+      - name: Set the list of image_variants for later jobs
+        id: image_variants
+        run: |
+          if [ "${{ github.event_name }}" == "pull_request" ] || [ "${{ github.ref_name }}" == "latest" ]; then
+            echo "image_variants=[\"nightly\",\"stable\"]" >> "$GITHUB_OUTPUT"
+          else
+            echo "image_variants=[\"stable\"]" >> "$GITHUB_OUTPUT"
+          fi
+      - name: Set the temporary image tag
+        run: |
+          temp_tag="${GITHUB_REF_NAME%/*}"
+          echo "Building $temp_tag"
+          echo "TEMP_TAG=${temp_tag}" >> $GITHUB_ENV
+      - uses: actions/checkout@v2
+        with:
+          # by default, it uses a depth of 1
+          # this fetches all history so that we can read each commit
+          fetch-depth: 0
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v1
+        with:
+          python-version: '3.8'
+      - name: Build images
+        run: |
+          podman version
+          buildah version
+          podman build --format docker --file images/Containerfile.core.base --tag pulp/base:${TEMP_TAG} .
+          podman build --format docker --file images/pulp_ci_centos/Containerfile --tag pulp/pulp-ci-centos:${TEMP_TAG} --build-arg FROM_TAG=${TEMP_TAG} .
+      # we use the docker format (default), even though it may not be the fastest,
+      # because it supports saving both images at once.
+      # However, it seems to export the common layers twice.
+      # We should look into whether its possible to export just pulp-ci-centos,
+      # and tag the base image manually.
+      - name: Save podman images to tarball
+        run: podman save -m -o base-images.tar pulp/base:${TEMP_TAG} pulp/pulp-ci-centos:${TEMP_TAG}
+      - name: Cache podman images
+        uses: actions/cache/save@v3
+        with:
+          key: base-images
+          path: base-images.tar
+          fail-on-cache-miss: true
```

Comment:
We actually want this job to fail if the step fails to upload. Because we do not want an old (by default, caches are preserved for upto 1 week) cache to be used.

Actually, the correct solution is to use a different cache key per CI run. [Instructions for this are not provided](https://github.com/actions/cache#creating-a-cache-key), but I can probably figure it out without too much effort.